File: 2024/july/javaScript/25.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArray = function(nums) {
    let mx = 5e4;
    let f = new Array(2 * mx + 1).fill(0);
    for (const cur of  nums) f[cur + mx]++;
    const n = nums.length;
    const ans = new Array(n).fill(0);
    let k = 0;
    for (let i = -mx; i <= mx; ++i){
        const sz = f[mx + i];
        for (let j = 0; j < sz; ++j) ans[k++] = i;
    }
    return ans; 
};

File: 2024/july/java/25.java
class Solution {
    static final int mx = (int)5e4;
    public int[] sortArray(int[] nums) {
        var f = new int [2 * mx+1];
        for (var cur : nums) f[cur + mx]++;
        final int n = nums.length;
        var ans = new int[n];
        int k = 0;
        for (int i = -mx; i <= mx; ++i){
            final int sz = f[mx + i];
            for (int j = 0; j < sz; ++j) ans[k++] = i;
        }
        return ans;
        
    }
}
